mri/operators/fourier/orc_wrapper.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# pySAP - Copyright (C) CEA, 2017 - 2018
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+"""
+This modules contains classes for Off-Resonance Correction (ORC)
+
+:Author: Guillaume Daval-Frérot <guillaume.davalfrerot@cea.fr>
+"""
+
+import numpy as np
+
+from ..base import OperatorBase
+from .utils import (compute_mfi_coefficients,
+                    compute_mti_coefficients,
+                    compute_svd_coefficients,
+                    compute_fsvd_coefficients)
+
+
+class ORCFFTWrapper(OperatorBase):
+    """ Off-Resonance Correction FFT Wrapper
+
+    This class is used to wrap any FFT operator and change it
+    into an off-resonance correction multi-linear operator using
+    the method described in :cite: `sutton2003` with different
+    choices of coefficients described in :cite: `fessler2005`.
+
+    """
+    def __init__(self, fourier_op, field_map, time_vec, mask,
+                 coefficients="svd", weights="full", L="auto", n_bins="auto"):
+        """ Initialize and compute multi-linear correction coefficients
+
+        Parameters
+        ----------
+        fourier_op: OperatorBase
+            Cartesian or non-Cartesian Fourier operator to wrap
+        field_map: numpy.ndarray
+            B0 field inhomogeneity map (in Hz)
+        time_vec: numpy.ndarray
+            1D vector indicating time after pulse in each shot (in s)
```

Comment:
Again, a utils function to get this time vector if we know the start of shot, the TE etc can be helpful. Possibly with defaults set for start time and TE as well.

Reply:
Similarly to the field map, it deeply depends on the trajectory and might not be as trivial as our usual case.